src/flow_solver/flow_solver.cpp

```diff
@@ -42,7 +42,11 @@ FlowSolver<dim, nstate>::FlowSolver(
 , dg(DGFactory<dim,double>::create_discontinuous_galerkin(&all_param, poly_degree, flow_solver_param.max_poly_degree_for_adaptation, grid_degree, flow_solver_case->generate_grid()))
 {
     flow_solver_case->set_higher_order_grid(dg);
-    dg->allocate_system();
+    if (flow_solver_param.steady_state == false) {
+        dg->allocate_system(false,false,false);
+    } else {
+        dg->allocate_system(true,true,true);
```

Comment:
We currently cannot compute dRdW, dRdX and second derivatives all at once. WeakDG uses codiPack and the corresponding functions (see, for example, [DGWeak::assemble_volume_term_derivatives()](https://github.com/dougshidong/PHiLiP/blob/f65960b8139e25f8e819a086261dbd605374d9f5/src/dg/weak_dg.cpp#L4130)) compute either first or second derivatives. For the flow solver, we just need residual derivative w.r.t. solution and this line can be replaced with `dg->allocate_system(true,false,false)`.

Reply:
The default is `(true,true,true)` (see `dg.h`) so that's what was being done before, should I still replace it by `(true,false,false)`?

Reply:
Technically flow solver just needs dRdW and it's best to only allocate it's memory. It might help prevent memory issues when we run implicit solver on large test cases.

Reply:
fixed